src/common/mclock_common.cc

```diff
@@ -380,80 +342,10 @@ uint32_t MclockConfig::calc_scaled_cost(int item_cost)
 void MclockConfig::handle_conf_change(const ConfigProxy& conf,
 				      const std::set<std::string> &changed)
 {
-  if (changed.count("osd_mclock_max_capacity_iops_hdd") ||
-      changed.count("osd_mclock_max_capacity_iops_ssd")) {
-   set_osd_capacity_params_from_config();
-   client_registry.update_from_config(
-      conf, osd_bandwidth_capacity_per_shard);
-  }
-  if (changed.count("osd_mclock_max_sequential_bandwidth_hdd") ||
-      changed.count("osd_mclock_max_sequential_bandwidth_ssd")) {
-    set_osd_capacity_params_from_config();
-    client_registry.update_from_config(
-      conf, osd_bandwidth_capacity_per_shard);
-  }
-  if (changed.count("osd_mclock_profile")) {
-    set_config_defaults_from_profile();
-    client_registry.update_from_config(
-      conf, osd_bandwidth_capacity_per_shard);
-  }
-
-  auto get_changed_key = [&changed]() -> std::optional<std::string> {
-    static const std::vector<std::string> qos_params = {
-      "osd_mclock_scheduler_client_res",
-      "osd_mclock_scheduler_client_wgt",
-      "osd_mclock_scheduler_client_lim",
-      "osd_mclock_scheduler_background_recovery_res",
-      "osd_mclock_scheduler_background_recovery_wgt",
-      "osd_mclock_scheduler_background_recovery_lim",
-      "osd_mclock_scheduler_background_best_effort_res",
-      "osd_mclock_scheduler_background_best_effort_wgt",
-      "osd_mclock_scheduler_background_best_effort_lim"
-    };
-
-    for (auto &qp : qos_params) {
-      if (changed.count(qp)) {
-        return qp;
-      }
-    }
-    return std::nullopt;
-  };
-  if (auto key = get_changed_key(); key.has_value()) {
-    auto mclock_profile = cct->_conf.get_val<std::string>("osd_mclock_profile");
-    if (mclock_profile == "custom") {
```

Comment:
This is great. Previously, when a user would set one of the mclock keys above on a **non-custom** profile - we had to remove the key set manually (_without_ letting the user know).
With the change here, we are able to ignore the set keys when a non-custom profile is set (Thanks to `current_profile` using the custom keys only when the profileis custom). 

@rzarzynski, with this we can drop the monc usage from the mclock code.

Reply:
>Would it help if the existing config options can be renamed to something like for e.g., osd_mclock_scheduler_client_res_custom and explicitly use it for custom profile only and make it clearer?

It makes sense to me. Additional suggestion here:
`osd_mclock_scheduler_custom_profile_client_res`